src/common/io.ts

```diff
@@ -950,10 +956,14 @@ export class ARIOWriteable extends ARIOReadable implements AoARIOWrite {
       super(config);
     }
     this.signer = createAoSigner(signer);
-    this.paymentProvider = new TurboArNSPaymentProvider({
-      signer: signer as Signer,
-      paymentUrl,
-    });
+    this.paymentProvider = isTurboArNSSigner(signer)
+      ? new TurboArNSPaymentProviderAuthenticated({
+          signer,
+          paymentUrl,
+        })
+      : new TurboArNSPaymentProviderUnauthenticated({
+          paymentUrl,
+        });
```

Comment:
i'd prefer to just have a factory utility or class that returns the signer vs. the ternary here

Reply:
[0527b83](https://github.com/ar-io/ar-io-sdk/pull/458/commits/0527b8357fe47f1dee8831832b43a520555a5770)